merkletree/node.go

```diff
@@ -7,28 +7,30 @@ import (
 
 type node struct {
 	parent MerkleNode
-	level  uint32
 }
 
 type interiorNode struct {
 	node
+	Level      uint32
 	leftChild  MerkleNode
 	rightChild MerkleNode
-	leftHash   []byte
-	rightHash  []byte
+	LeftHash   []byte
+	RightHash  []byte
 }
 
 type userLeafNode struct {
 	node
-	key        string
-	value      []byte
-	index      []byte
-	commitment *crypto.Commit
+	Level      uint32
+	Key        string
+	Value      []byte
+	Index      []byte
+	Commitment *crypto.Commit
```

Comment:
Yes, since we didn't export the structs so I think it still ensures our preservation, i.e., there's no way to change or read the node's fields.
Btw, the reason is (mostly) because I didn't want to implement the GobEncode/GobDecode interface manually.